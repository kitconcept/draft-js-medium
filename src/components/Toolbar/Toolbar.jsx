/**
 * Toolbar component.
 * @module components/Toolbar
 */

import React, { Component } from 'react';

import Button from '../Button/Button';

/**
 * Toolbar component class.
 * @class Toolbar
 * @extends Component
 */
export default class Toolbar extends Component {
  /**
   * Render method.
   * @function render
   * @returns {string} Markup of the container.
   */
  render() {
    return (
      <div className="ui black pointing below label">
        <div className="ui black buttons">
          <Button icon="bold" />
          <Button icon="italic" />
          <Button icon="linkify" />
        </div>
      </div>
    );
  }
}
