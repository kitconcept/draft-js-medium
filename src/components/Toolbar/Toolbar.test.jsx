import React from 'react';
import renderer from 'react-test-renderer';
import { render } from 'enzyme';

import Toolbar from './Toolbar';

jest.mock('../Button/Button', () => jest.fn(() => <div className="button" />));
jest.mock('../StyleButton/StyleButton', () =>
  jest.fn(() => <div className="style-button" />),
);

describe('Toolbar', () => {
  it('renders a toolbar component', () => {
    const component = renderer.create(
      <Toolbar onToggleInline={() => {}} onToggleLink={() => {}} selection />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders an empty span when no selection is specified', () => {
    const component = renderer.create(
      <Toolbar
        onToggleInline={() => {}}
        onToggleLink={() => {}}
        selection={false}
      />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  /*
  it('repositions the toolbar on update', () => {
    const wrapper = render(
      <Toolbar
        onToggleInline={() => {}}
        onToggleLink={() => {}}
        selection={false}
      />,
    );
    wrapper.setProps({ selection: true });
  });
  */
});
