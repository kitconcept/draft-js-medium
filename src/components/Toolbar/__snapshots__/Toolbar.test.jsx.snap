// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar renders a toolbar component 1`] = `
<div
  className="ui black pointing below label"
  style={
    Object {
      "position": "absolute",
    }
  }
>
  <div
    className="ui black buttons"
  >
    <div
      className="style-button"
    />
    <div
      className="style-button"
    />
    <div
      className="button"
    />
  </div>
</div>
`;

exports[`Toolbar renders an empty span when no selection is specified 1`] = `<span />`;
